### PR TITLE
header apihandler added

### PR DESCRIPTION
**Pull request by younger15**

### CODE REVIEW

File: src/Header/APIHandler.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class APIHandler
+{
+
+public:
+    virtual void CallingAPI(OperateData data);
+
+private:
+    OperateData response;
+    ResponseHandler responseHandle;
+};
```
Comment (john0312) [inline src/Header/APIHandler.h]: New line at the end.

File: src/Header/APIHandler.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <string>
+#include "./OperateData.h"
+
+class APIHandler
+{
+
+public:
```
Comment (john0312) [inline src/Header/APIHandler.h:15]: 1. Missing constructor.
2. Is this copyable? If not, please delete copy constructor.

File: src/Header/APIHandler.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    virtual void CallingAPI(OperateData data);
```
Comment (john0312) [inline src/Header/APIHandler.h]: Please add comment for member method.

File: src/Header/APIHandler.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+private:
+    OperateData response;
```
Comment (john0312) [inline src/Header/APIHandler.h:24]: Please add comment for member variable.

File: src/Header/APIHandler.h (hunk 3 above)
Comment (john0312) [reply src/Header/APIHandler.h]: Use const reference for input.

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
```
Comment (john0312) [inline LICENSE]: LICENSE should be committed as a separate commit.

**Review by john0312 — COMMENTED** [review]
Please squash revert changes so they don't appear in the pull request.

File: src/Header/APIHandler.h (hunk 1 above)
Comment (younger15) [reply src/Header/APIHandler.h]: new line added

File: src/Header/APIHandler.h (hunk 2 above)
Comment (younger15) [reply src/Header/APIHandler.h:15]: noncopyable  constructor added

File: src/Header/APIHandler.h (hunk 3 above)
Comment (younger15) [reply src/Header/APIHandler.h]: comment added and parameter changed

File: LICENSE (hunk 5 above)
Comment (younger15) [reply LICENSE]: There exist another LICENSE file, this is one kind of LICENSE format that mentor told us

File: src/Header/APIHandler.h [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+{
+
+public:
+    APIHandler();
+    //the start entry of all api services
+    virtual void CallingAPI(const OperateData &data);
+
+private:
+    //non copyable
+    APIHandler(const APIHandler&);
```
Comment (john0312) [inline src/Header/APIHandler.h]: Please use =delete

File: src/Header/APIHandler.h [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    APIHandler& operator=(const APIHandler&);
+    //will generate response after service end or an error occured.
+    OperateData response;
+    ResponseHandler responseHandle;
```
Comment (john0312) [inline src/Header/APIHandler.h]: Please add comment for member variable.

File: src/Header/APIHandler.h [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+#endif
```
Comment (john0312) [inline src/Header/APIHandler.h]: Please add newline.

File: src/Header/APIHandler.h (hunk 7 above)
Comment (younger15) [reply src/Header/APIHandler.h]: Done, variable removed, handler will be a singleton so doesn't need class variable

File: src/Header/APIHandler.h (hunk 8 above)
Comment (younger15) [reply src/Header/APIHandler.h]: Done

File: src/Header/APIHandler.h (hunk 6 above)
Comment (younger15) [reply src/Header/APIHandler.h]: Done

File: src/Header/APIHandler.h (hunk 4 above)
Comment (younger15) [reply src/Header/APIHandler.h:24]: Done

**Review by john0312 — APPROVED** [review]
LGTM, approved.
